### DISCUSSION TITLE
How can I indent a Rich Tree?

### DISCUSSION THREAD

**Discussion by ncoish** [Textualize/rich#1642] [Q&A] [answered]
Hi all,

This library is incredible, but there's one thing I haven't been able to figure out without doing some real hackery:

I'm trying to render a tree as part of a formatted preamble to a CLI that I'm developing. Right now the rest of the preamble has a number of indentation levels, and I would like the tree to be at the same indentation as the rest of the test where it occurs, instead of being pinned to the far left of the console.

Please forgive me if this is something that I'm missing from the docs, but is there a way to set the indentation of a tree (or any other renderable?)

**Comment by willmcgugan** [comment, marked as answer]
You can use the Padding class for this. There is a shortcut that adds padding to the left. Here's an example:

```
>>> from rich.padding import Padding
>>> from rich import print
>>> from rich.tree import Tree
>>> tree = Tree("foo")
>>> tree.add("bar")
<rich.tree.Tree object at 0x7feb2c5347f0>
>>> print(tree)
foo
└── bar
>>> print(Padding.indent(tree, 5))
     foo
     └── bar
```

**Reply by ncoish** [reply]
This does exactly what I was looking for! For some reason when I read the Padding docs I missed indent entirely: I thought it could only add vertical padding.

Thank you very much!
